packages/validator-bonds-sdk/__tests__/bankrun/bankrun.ts

```diff
@@ -98,3 +119,54 @@ export async function delegateAndFund({
     validatorIdentity,
   }
 }
+
+export enum StakeActivationState {
+  Activating,
+  Deactivating,
+  Activated,
+  Deactivated,
+  Unknown,
+  NonDelegated,
+}
+
+export async function stakeActivation(
+  provider: BankrunExtendedProvider,
+  stakeAccount: PublicKey
+): Promise<StakeActivationState> {
+  const [stakeState] = await getAndCheckStakeAccount(
+    provider,
+    stakeAccount,
+    StakeStates.Delegated
+  )
+  if (stakeState.Stake !== undefined) {
+    const activationEpoch = stakeState.Stake.stake.delegation.activationEpoch
+    const deactivationEpoch =
+      stakeState.Stake.stake.delegation.deactivationEpoch
+    const curEpoch = new BN(await currentEpoch(provider))
+    console.log(
+      'activationEpoch',
+      activationEpoch.toString(),
+      'deactivationEpoch',
+      deactivationEpoch.toString(),
+      'currentEpoch',
+      curEpoch.toString()
+    )
+
+    if (!deactivationEpoch.eq(U64_MAX) && deactivationEpoch.gte(curEpoch)) {
+      return StakeActivationState.Deactivating
```

Comment:
activationEpoch = 500
deactivationEpoch = 500
---> for this case the correct state would be Deactivated